### PR TITLE
Add 'partial' annotations

### PR DESCRIPTION
**Pull request by ohad**
Since the compiler can't prove these functions total

### CODE REVIEW

**Comment by gallais** [comment]
Note that `covering` should be enough.

File: Idrall/Check.idr [hunk 1]
```diff
@@ -150,6 +150,7 @@ vAnyPi t = Left $ Unexpected (getFC t) $ show t ++ " is not a VPi or VHPi"
 data Types = TEmpty
            | TBind Types Name Value
 
+partial
```
Comment (ohad) [inline Idrall/Check.idr:153]:
```suggestion
covering
```

File: Idrall/Check.idr [hunk 2]
```diff
@@ -175,6 +176,7 @@ record Cxt where
   values : Env
   types  : Types
 
+partial
```
Comment (ohad) [inline Idrall/Check.idr:179]:
```suggestion
covering
```

File: Idrall/Expr.idr [hunk 3]
```diff
@@ -301,14 +301,14 @@ Show (Import ImportStatement) where
   show (Resolved x) = "(Resolved x)"
 
 mutual
-  export
+  export partial
```
Comment (ohad) [inline Idrall/Expr.idr:304]:
```suggestion
  export covering
```

File: Idrall/Expr.idr [hunk 4]
```diff
@@ -301,14 +301,14 @@ Show (Import ImportStatement) where
...
   Show (Import a) where
     show (Raw x) = "(Raw)" -- TODO show x
     show (Text x) = "(Text)" -- TODO show x
     show (Location x) = "(Location)" -- TODO show x
     show (Resolved x) = "(Resolved " ++ show x ++ ")"
 
-  export
+  export partial
```
Comment (ohad) [inline Idrall/Expr.idr:311]:
```suggestion
  export covering
```

File: Idrall/Expr.idr [hunk 5]
```diff
@@ -382,7 +382,7 @@ mutual
     show (EImportAlt fc x y) = "(EImportAlt " ++ show x ++ " " ++ show y ++ ")"
     show (EEmbed fc x) = "(EEmbed " ++ show x ++ ")"
 
-  public export
+  public export partial
```
Comment (ohad) [inline Idrall/Expr.idr:385]:
```suggestion
  public export covering
```

File: Idrall/Value.idr [hunk 6]
```diff
@@ -126,27 +126,28 @@ VPrim : (Value -> Either Error Value) -> Value
...
 
 mutual
+  partial
   Show HLamInfo where
     show Prim = "Prim"
     show (Typed x y) = "(Typed " ++ show x ++ " " ++ show y ++ ")"
     show NaturalSubtractZero = "NaturalSubtractZero"
 
-  public export
+  public export partial
```
Comment (ohad) [inline Idrall/Value.idr:135]:
```suggestion
  public export covering
```

File: Idrall/Value.idr [hunk 7]
```diff
@@ -126,27 +126,28 @@ VPrim : (Value -> Either Error Value) -> Value
...
   Show Env where
     show Empty = "Empty"
     show (Skip x y) = "(Skip " ++ show x ++ " " ++ show y ++ ")"
     show (Extend x y z) = "(Extend " ++ show x ++ " " ++ show y ++ " " ++ show z ++ ")"
 
-  public export
+  public export partial
```
Comment (ohad) [inline Idrall/Value.idr:141]:
```suggestion
  public export covering
```

File: Idrall/Value.idr [hunk 8]
```diff
@@ -126,27 +126,28 @@ VPrim : (Value -> Either Error Value) -> Value
...
   Show Closure where
     show (MkClosure closureName closureEnv closureBody)
       = "(MkClosure " ++ show closureName ++ " " ++ show closureEnv ++ " " ++ show closureBody ++ ")"
 
-  public export
+  public export partial
```
Comment (ohad) [inline Idrall/Value.idr:146]:
```suggestion
  public export covering
```

File: Idrall/Value.idr [hunk 9]
```diff
@@ -126,27 +126,28 @@ VPrim : (Value -> Either Error Value) -> Value
...
   Show VChunks where
     show (MkVChunks xs x) = "(MkVChunks " ++ show xs ++ " " ++ show x ++ ")"
 
-  public export
+  public export partial
```
Comment (ohad) [inline Idrall/Value.idr:150]:
```suggestion
  public export covering
```

**Comment by ohad** [comment]
Fixed now!

**Pull request merged**